nifi-nar-bundles/nifi-azure-bundle/nifi-azure-record-sink/src/main/java/org/apache/nifi/services/azure/eventhub/AzureEventHubRecordSink.java

```diff
@@ -0,0 +1,229 @@
+    public static final PropertyDescriptor SHARED_ACCESS_POLICY_KEY = new PropertyDescriptor.Builder()
+            .name("Shared Access Policy Key")
+            .description("The primary key of the shared access policy")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .sensitive(true)
+            .required(false)
+            .dependsOn(AUTHENTICATION_STRATEGY, AzureAuthenticationStrategy.DEFAULT_AZURE_CREDENTIAL.getValue())
+            .build();
+    public static final PropertyDescriptor PARTITION_KEY = new PropertyDescriptor.Builder()
+            .name("Partition Key")
+            .description("A hint for Eventhub message broker how to distribute messages consistently amongst multiple partitions.")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .required(false)
+            .build();
+    private static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = Collections.unmodifiableList(
+            Arrays.asList(
+                    EVENT_HUB_NAME,
+                    NAMESPACE,
+                    SERVICE_BUS_ENDPOINT,
+                    SHARED_ACCESS_POLICY,
+                    SHARED_ACCESS_POLICY_KEY,
+                    AUTHENTICATION_STRATEGY,
+                    PARTITION_KEY,
+                    RECORD_WRITER_FACTORY
+            )
+    );
+
+    private volatile ConfigurationContext context;
+    private RecordSetWriterFactory writerFactory;
+    private EventHubProducerClient client;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    protected EventHubProducerClient createEventHubClient(final String namespace, final String serviceBusEndpoint, final String eventHubName,
+                                                          final String policyName, final String policyKey, final AzureAuthenticationStrategy authenticationStrategy) throws ProcessException {
+        final AzureNamedKeyCredential azureNamedKeyCredential = new AzureNamedKeyCredential(policyName, policyKey);
+        final String fullyQualifiedNamespace = String.format("%s%s", namespace, serviceBusEndpoint);
+        final DefaultAzureCredentialBuilder defaultAzureCredentialBuilder = new DefaultAzureCredentialBuilder();
+        final DefaultAzureCredential defaultAzureCredential = defaultAzureCredentialBuilder.build();
+        final EventHubClientBuilder eventHubClientBuilder = new EventHubClientBuilder();
+        if (AzureAuthenticationStrategy.SHARED_ACCESS_KEY == authenticationStrategy) {
+            eventHubClientBuilder.credential(fullyQualifiedNamespace, eventHubName, azureNamedKeyCredential);
+        } else {
+            eventHubClientBuilder.credential(fullyQualifiedNamespace, eventHubName, defaultAzureCredential);
+        }
```

Comment:
The credential objects should be built inside the conditionals.
```suggestion
        final EventHubClientBuilder eventHubClientBuilder = new EventHubClientBuilder();
        if (AzureAuthenticationStrategy.SHARED_ACCESS_KEY == authenticationStrategy) {
            final AzureNamedKeyCredential azureNamedKeyCredential = new AzureNamedKeyCredential(policyName, policyKey);
            eventHubClientBuilder.credential(fullyQualifiedNamespace, eventHubName, azureNamedKeyCredential);
        } else {
            final DefaultAzureCredentialBuilder defaultAzureCredentialBuilder = new DefaultAzureCredentialBuilder();
            final DefaultAzureCredential defaultAzureCredential = defaultAzureCredentialBuilder.build();
            eventHubClientBuilder.credential(fullyQualifiedNamespace, eventHubName, defaultAzureCredential);
        }
```